TargetComposition/SymBuilder/CallingConvention.h

```diff
@@ -0,0 +1,131 @@
+//**************************************************************************
+//
+// CallingConvention.h
+//
+// The header for our understanding of calling conventions.
+//
+// NOTE: This particular component is on the cusp between the target model side of the
+//       extension and the data model side.  It throws exceptions and that takes a great
+//       deal of care with its usage on the target model side.
+//
+//**************************************************************************
+//
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+//
+//**************************************************************************
+
+#ifndef __CALLINGCONVENTION_H__
+#define __CALLINGCONVENTION_H__
+
+namespace Debugger
+{
+namespace TargetComposition
+{
+namespace Services
+{
+namespace SymbolBuilder
+{
+
+// CallingConvention:
+//
+// An abstraction of a calling convention.
+//
+class CallingConvention
+{
+public:
+
+    // CallingConvention():
+    //
+    // Initializes a calling convention object.
+    //
+    CallingConvention(_In_ SymbolBuilderManager *pManager,
+                      _In_ size_t numNonVolatiles,
+                      _In_reads_(numNonVolatiles) wchar_t const **ppNonVolatileNames);
+
+    // GetParameterPlacements():
+    //
+    // Given a list of parameters to a function, fill in their locations with knowledge of the calling
+    // convention.
+    //
+    virtual void GetParameterPlacements(_In_ size_t paramCount,
+                                        _In_reads_(paramCount) VariableSymbol **ppParameters,

```

Comment:
Should this be _In_reads_(paramCount) VariableSymbol const **ppParameters ?

Reply:
What do you mean...?  It *IS* annotated _In_reads_(paramCount).  The method reads the information about the 'paramCount' parameters in ppParameters and fills in their initial locations according to the calling convention.

Reply:
Discussed offline :)

Reply:
Per offline commentary -- made this "VariableSymbol const **" and added necessary requisite upstream changes.